### PR TITLE
Note PSF licence in PyPI metadata

### PR DESCRIPTION
**Pull request by hugovk**
As mentioned in the `README` and classifiers, this project is using the PSF license, so let's update the metadata in `pyproject.toml` to match.

Also:

* Fix some typos and formatting
* Upgrade syntax for Python 3.7+
* More f-strings
* Ignore `build` dir from Git

Let me know if any of these should be split out.

### CODE REVIEW

File: sphinxlint.py [hunk 1]
```diff
@@ -518,16 +508,14 @@ def main(argv=None):
...
         else:
             print("No problems found.")
     else:
         for severity in sorted(count):
             number = count[severity]
-            print(
-                "%d problem%s with severity %d found."
-                % (number, number > 1 and "s" or "", severity)
-            )
+            s = number > 1 and "s" or ""
```
Comment (JulienPalard) [inline sphinxlint.py:517]: Why not:

```suggestion
            s = "s" if number > 1 else ""
```

**Comment by JulienPalard** [comment]
Thanks!!

> Let me know if any of these should be split out.

No that's OK no worries, thanks a lot for spotting so much, looks like I have to work my english :D

File: sphinxlint.py (hunk 1 above)
Comment (hugovk) [reply sphinxlint.py:517]: Much more readable, thanks!

**Review by JulienPalard — APPROVED** [review]

**Comment by JulienPalard** [comment]
Thanks!

**Pull request merged**